packages/driver/cypress/e2e/commands/cookies.cy.js

```diff
@@ -2,21 +2,317 @@ const { assertLogLength } = require('../../support/utils')
+
+      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+
+      cy.origin('http://www.foobar.com:3500', () => {
+        cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
+
+        cy.getCookies({ domain: null }).should('have.length', 2)
+      })
+    })
+  })
+
+  context('#getCookie', () => {
+    it('returns the cookie from the domain matching the AUT by default', () => {
+      cy.visit('http://localhost:3500/fixtures/generic.html')
+      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
+      cy.setCookie('foo', 'bar')
+
+      cy.getCookie('foo').its('domain').should('equal', 'localhost')
+
+      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+
+      cy.origin('http://www.foobar.com:3500', () => {
+        cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
+
+        cy.getCookie('foo').its('domain').should('match', /\.?www\.foobar\.com/)
+      })
+    })
+
+    it('returns the cookie from the specified domain', () => {
+      cy.visit('http://localhost:3500/fixtures/generic.html')
+      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
+      cy.setCookie('foo', 'bar')
+
+      cy.getCookie('foo', { domain: 'www.foobar.com' })
+      .its('domain').should('match', /\.?www\.foobar\.com/)
+
+      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+
+      cy.origin('http://www.foobar.com:3500', () => {
+        cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
+
+        cy.getCookie('foo', { domain: 'localhost' })
+        .its('domain').should('equal', 'localhost')
+      })
+    })
+
+    it('returns cookie for any domain when domain is null', () => {
+      cy.visit('http://localhost:3500/fixtures/generic.html')
+      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
+
+      cy.getCookie('foo', { domain: null })
+      .its('domain').should('match', /\.?www\.foobar\.com/)
+
+      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+
+      cy.origin('http://www.foobar.com:3500', () => {
+        cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
+
+        cy.clearCookie('foo')
+        cy.setCookie('foo', 'bar', { domain: 'localhost' })
+
+        cy.getCookie('foo', { domain: null })
+        .its('domain').should('equal', 'localhost')
+      })
+    })
+  })
+
+  context('#setCookie', () => {
+    it('sets the cookie on the domain matching the AUT by default', () => {
+      cy.visit('http://localhost:3500/fixtures/generic.html')
+      cy.setCookie('foo', 'bar')
+
+      cy.getCookie('foo').its('domain').should('equal', 'localhost')
+
+      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+
+      cy.origin('http://www.foobar.com:3500', () => {
+        cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
+        cy.setCookie('foo', 'bar')
+
+        cy.getCookie('foo').its('domain').should('equal', '.foobar.com')
```

Comment:
The default domain is the superdomain of the AUT, which is existing behavior for all cookie commands, so this is correct. I don't know of a story to change that, but it would be a breaking change. I think it should be a separate PR if we do decide to do that, since this can go straight into a release as-is.

Reply:
Agreed. I'm thinking we should do this work to be consistent with cookie behavior, but not into `develop` since it does break the API. Just want to make sure we capture that someplace. I created #24526 and added it into the next sprint to target `v12.0.0`

Reply:
Yeah it's just changing [this one line](https://github.com/cypress-io/cypress/blob/b595636d061d138cccc7d9a1e5f965145e995a6b/packages/driver/src/cy/commands/cookies.ts#L35) plus updating tests as needed.